### ISSUE TITLE
design "Lite AR Mode" that doesn't require permissions/consent equivalent to camera

### ISSUE THREAD

**Issue by lincolnfrog** [immersive-web/webxr#402]
This is important if we want to support a safer version of AR that people will use on less-trusted areas of the web. Somehow we need to restrict what information (geometry/etc) is exposed so that the user has a strong guarantee of privacy of their data to the AR app. This is, of course, heavily related to https://github.com/immersive-web/privacy-and-security.

It also is a replacement for issues such as https://github.com/immersive-web/hit-test/issues/6. The reasoning is that since apps can be creative with their use of an API like hit-test, it is basically impossible for us to use rate-limiting, ray-limiting, or user-gesture-limiting to determine what geometry will be exposed. Instead, we should have a spec for this lite mode limiting what underlying geometry is considered for APIs such as hit-test. Hit-test can be unlimited if it can only fundamentally interact with a user-chosen plane on the floor in front of them, for example.

**Comment by tencircles** [comment]
As I brought up in the F2F, I think the fundamental question here is what motivation will developers have to use lite-mode if both lite-mode and heavy-mode require a native permissions dialogue given that both will likely elicit similar rejection rates? 

@toji mentioned that a simple notification (as with fullscreen mode) might suffice for lite-mode, but @johnpallett mentioned that the Chrome security team would require a permissions dialog for any world understanding.

I think there is a good balance to be struck between AR Quicklook and giving full camera and mesh access, but I'm not exactly sure where that is. There definitely is a subset of use cases that could be supported by a mode with no/limited world understanding, but the question then is: do these use cases support the creation of a separate mode? Thoughts?

**Comment by johnpallett** [comment]
@tencircles thanks for the comment! To clarify one detail, a permissions dialog is only one approach to getting user consent, but other forms of user consent such as a dedicated [AR Mode](https://github.com/immersive-web/privacy-and-security/pull/11) could also work. 

Whatever the approach, you're correct that we'd want user consent before any data could be shared with a site.

**Comment by NellWaliczek** [comment]
Hey Max, thanks for filing this!  @avadacatavra (Diane) at Mozilla is interested in working on this with you.  Please reach out to each other and I'm looking forward to hearing a proposal!

**Comment by avadacatavra** [comment]
I think that we should start by considering the differences between an AR-Lite and Immersive experience--from there, we can identify potential threat vectors and mitigations and evaluate if it's a useful middle ground.

My understanding of the basic premise of AR-Lite mode is:
* the browser composites the scene using pixels provided from the site
* any environment data is isolated to the browser and not transmitted to the site

**Comment by cwilso** [comment]
I feel like maybe this should move to /proposals?

**Comment by NellWaliczek** [comment]
Agreed that this might make more sense in the proposals repo.  However, I'd like to us have a rough idea of whether or not an AR Lite mode is even meaningful by the end of the year.  

To be clear, I'm not suggesting we need final API designs by then.  Rather, I'd like to see a reporting on the experimentation proposed at the Sept '19 F2F to determine whether or not a subset of functionality would actually be valuable to developers and end users.  From my recollection, this included experiments about how much data could be reconstructed from variants of rate-limited hit testing and experiments regarding how to communicate the difference in permissions to end users in a way that was actually comprehensible.

The date is somewhat arbitrary, but it comes from my concern that if we don't closure on the high-level value/feasibility by end of this year, I strongly suspect this feature will not make the CR of WebXR 1.0.

**Comment by NellWaliczek** [comment]
On the technical front, I also wanted to put a thought out there...

I haven't thought through the threat vectors of this, but pulling together a number of different threads of conversation I heard at the F2F, one potential starting point for the technical details (not the user consent part) might look something like:
- Add a function to enable automatic occlusion based on a depth buffer (probably the depth buffer would be supplied via a different method because there's other value in having one too)
- Add a function which allows the developer to supply a 3D object for the UA to draw at the "cursor" location.  The developer decides what this should be so it could be a simple dot, or it could be a preview of an object that will be anchored at the hit-tested location at the time a user gesture completes.  Obviously this opens a big can of worms regarding the UA needing to do any 3D rendering of its own.
- Do not allow arbitrary hit-test matrices to be requested by the page
- In response to a user gesture (such as select up or select down; and yes we'd need to make this work for other controller buttons, etc.) allow an XRAnchor to be created at the related XRInputSource's pointing pose's hit-test result (+ an offset if needed).  This does not need to expose the matrices of the hit test result to the page, though I'm not sure it matters for security reasons since the XRAnchor's relative transform can always be computed subsequently.

**Comment by lincolnfrog** [comment]
@NellWaliczek @avadacatavra Happy to collaborate on defining this better.

**Comment by NellWaliczek** [comment]
Awesome.  <3
If there's no objection the sense of urgency I alluded to above, then I'm ok with moving this into proposals for next steps.  @lincolnfrog that ok?  If so, @TrevorFSmith can you migrate this issue?

**Comment by blairmacintyre** [comment]
I'd be happy to be involved in this too.
